File: Back-End/src/main/java/pt/luis/blogapp/api/services/commentService/CommentService.java
package pt.luis.blogapp.api.services.commentService;

import org.springframework.data.domain.Page;
import org.springframework.http.ResponseEntity;
import pt.luis.blogapp.api.dto.commentDTO.CreateCommentDTO;
import pt.luis.blogapp.api.dto.commentDTO.ResponseCommentDTO;
import pt.luis.blogapp.api.dto.commentDTO.UpdateCommentDTO;

import java.util.List;

public interface CommentService {

    ResponseCommentDTO create(CreateCommentDTO dto);

    List<ResponseCommentDTO> getByAuthor(String author);
    Page<ResponseCommentDTO> getAllPaged(
            Long postId,
            int page,
            int size,
            String sortBy,
            String direction
    );
    ResponseCommentDTO update(Long id, UpdateCommentDTO dto);
    void delete(Long id);
}

File: Front-End/my-react-app/src/services/CommentService.ts
import { api } from "../api/api"
import { Comment } from "../types/comment/Comment";


export const CommentService ={

    async getByPost(postId:number): Promise <Comment[]>{
        const response = await api.get("/comments",{
            params:{
                postId,
                page:0,
                size:50,
                sortBy:"createdAt",
                direction:"desc"
            }
        })

        return response.data.content;
    },

    async create(comment:{content:string, postId:number}){
        const response = await api.post("/comments", comment);
        return response.data;
    },

    async update(id:number, content:string){
        const response = await api.put(`/comments/${id}`, {content});
        return response.data;
    },

    async delete(id:number){
        await api.delete(`/comments/${id}`);
    }
}